tka/tailchonk.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package tka
+
+import (
+	"os"
+	"sync"
+)
+
+// Chonk implementations provide durable storage for AUMs and other
+// TKA state.
+//
+// All methods must be thread-safe.
+//
+// The name 'tailchonk' was coined by @catzkorn.
+type Chonk interface {
+	// GetAUM returns the AUM with the specified digest.
+	//
+	// If the AUM does not exist, then os.ErrNotExist is returned.
```

Comment:
os.ErrNotExist is an alias these days for fs.ErrNotExist, which requires fewer dependencies.

I'm not sure it's exactly the error the error to use, but I guess it's ok for now.

Reply:
I'm hoping that LTO is smart enough to remove anything that doesn't get used, but in any case I'm guessing the os package is pulled in anyway?

If it ends up mattering, I can make a specific `ErrNotExist` error for this package.